lib/Remotery.c

```diff
@@ -789,6 +802,14 @@ static rmtU32 GaloisLFSRNext(rmtU32 value, rmtU32 xor_mask)
         }                                                      \
     }
 
+// New will allocate enough space for the object and call the constructor
+// If allocation fails the constructor won't be called
+// If the constructor fails, the destructor is called and memory is released
+// NOTE: Use of sizeof() requires that the type be defined at the point of call
+// This is a disadvantage over requiring only a custom Create function
+#define rmtTryNew(...)                                         \
+    _VA_ARGS_SELECT(_rmtTryNew,__VA_ARGS__,N,N,N,1,1)(__VA_ARGS__)
```

Comment:
It almost compiles now.
The remaining issue is that not all functions have or return an `error` value.